src/main/java/core/basesyntax/actions/PurchaseAction.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.actions;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseAction implements DoingAction {
+    @Override
+    public Fruit getCount(FruitTransaction transaction, Fruit fruit) {
```

Comment:
rename this method (here and all other classes that implements DoingAction) .
name it like `applyOretation()`, or `applyAction()`. method should be void (no returns the result) and should save the result of the operation to the Storage class.